src/libcollections/str.rs

```diff
@@ -1297,6 +1301,899 @@ pub trait StrExt: Index<FullRange, Output = str> {
     }
 }
 
+
+/// Any string that can be represented as a slice.
+#[cfg(not(stage0))]
+#[stable]
+pub trait StrExt: Index<RangeFull, Output = str> {
```

Comment:
Instead of duplicating, could this do something like:

```rust
#[cfg(stage0)]
use core::ops::FullRange as RangeFull;
#[cfg(not(stage0))]
use core::ops::RangeFull;
```